ccube.nim

```diff
@@ -66,142 +66,101 @@ Examples:
+            quit()
+        result.number = val
+      of "file":
+        result.filePath = val
+    else: discard
 
-proc isEven(number:BigInt):bool =
-  if number mod 2 == 0:
-    result = true
-  else:
-    result = false
+  if result.filePath.len == 0:
+    result.filePath = getCurrentDir() / "iterationData.txt"
 
-proc readJson():(string,string,string) =
-  var isFile = fileExists("ccube_conf.json")
+  if result.number.len == 0:
+    result.number = $rand(1..1000)
+
+proc isEven(number: BigInt): bool =
+  result = number mod 2 == 0
+
+proc readJson(): (ProgramArgs, bool) =
+  var isFile = fileExists("ccube_config.json")
   if isFile:
     try:
-      let
-        readJsonData = readFile("ccube_conf.json")
-        jsonNode = parseJson(readJsonData)
-        noLoopSwitchBool = $jsonNode["noLoopSwitch"]
-        number = $jsonNode["number"]
-        filePath = $jsonNode["filePath"]
-      result = (noLoopSwitchBool,number,filePath)
+      let args = parseFile("ccube_config.json").to(ProgramArgs)
+      result = (args, true)
     except KeyError:
-      stderr.write("Error:An error occured while reading json keys,Invalid Keys!\n")
-      result = ("-1","-1","-1")
-  else:
-    result = ("-1","-1","-1")
+      stderr.write("An error occured while reading JSON configuration.")
 
-
-proc main(cmdArgs:(string,string,string)):void = 
+proc doWork(args: ProgramArgs) =
   var
-    noLoopSwitchBool = cmdArgs[0].strip(chars = {'"'})   # strip " from string if exists
-    number = initBigInt(cmdArgs[1].strip(chars = {'"'}))
-    filePath = cmdArgs[2].strip(chars = {'"'})
+    number = initBigInt(args.number)
     isLooping = true
-    incrementer:BigInt
-    noOfIterations:BigInt = initBigInt("0")
-  while isLooping or noLoopSwitchBool == "false":
+    incrementer: BigInt
+    noOfIterations = initBigInt("0")
+  
+  while isLooping:
     incrementer = number
     while number != 1:
       if isEven(number):
         number = number div 2
-        inc noOfIterations
       else:
         number = (number * 3) + 1
-        inc noOfIterations
+      inc noOfIterations
 
     var echoData = &"""-----------------------------------
-Current Number:{incrementer}
+Current number:{incrementer}
 Number of iterations:{noOfiterations}
 -----------------------------------{'\n'}"""
 
-    echo(echoData)
-    if filePath == "void":
-      discard
-    else:
+    echo echoData
+    if args.filePath != "void":
       try:
-        let writeData = open(filePath,fmAppend)
+        let writeData = open(args.filePath, fmAppend)
         writeData.write(echoData)
         writeData.close
       except IOError as e:
-        stderr.write(fmt"Error:An Error occured while writing data,{'\n'}{'\n'}Details:{e.msg}")
+        stderr.write(&"An error occured while writing data.\n\nDetails: {e.msg}")
         quit()
     inc incrementer
     number = incrementer
     noOfIterations = initBigInt("0")
-    if noLoopSwitchBool == "true":
+    if args.noLoopSwitch:
```

Comment:
At the end of the first iteration we just check if the noLoopSwitch is enabled, and if it is - stop the loop. You can also use `break` instead of `isLooping` here